src/System.Buffers.Experimental/System/Buffers/Text/BufferWriter_strings.cs

```diff
@@ -0,0 +1,200 @@
+            if (!format.TryTransform(Free, ref written)) return false;
+            _written += written;
+            return true;
+        }
+
+        public void WriteLine(Utf8Span text, TransformationFormat transformation)
+        {
+            while (!TryWriteLine(text, transformation)) Resize();
+        }
+
+        public bool TryWrite(Utf8String text)
+            => TryWrite((Utf8Span)text);
+
+        public void Write(Utf8String text)
+            => Write((Utf8Span)text);
+
+        public bool TryWrite(Utf8String text, TransformationFormat transformation)
+            => TryWrite((Utf8Span)text, transformation);
+
+        public void Write(Utf8String text, TransformationFormat transformation)
+            => Write((Utf8Span)text, transformation);
+        #endregion
+
+        public void Write(char character)
+        {
+            if (character <= 127)
+            {
+                var free = Free;
+                if (free.Length < 1)
+                {
+                    Resize();
+                    free = Free;
+                }
+                free[0] = (byte)character;
+                _written++;
+            }
+            else
+            {
+                Span<byte> utf8Span = stackalloc byte[4];
+                Span<char> utf16Span = stackalloc char[1];
+                utf16Span[0] = character;
+                if(Encodings.Utf8.FromUtf16(utf16Span.AsBytes(), utf8Span, out int consumed, out int written) == OperationStatus.Done)
+                {
+                    var encoded = utf8Span.Slice(0, written);
+                    while (!encoded.TryCopyTo(Free))
+                    {
+                        Resize();
+                    }
+                    _written += written;
+                }
+            }
+        }
+
+        private bool TryWrite(string text, out int written)
+        {
+            var status = Encodings.Utf16.ToUtf8(text.AsReadOnlySpan().AsBytes(), Free, out var consumed, out written);
+
+            switch (status)
+            {
+                case OperationStatus.Done:
+                    return true;
+                case OperationStatus.DestinationTooSmall:
+                    written = 0;
+                    return false;
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(text));
+            }
+        }
+
+        private bool TryWriteLine(string text, out int written)
+        {
+            if (!TryWrite(text, out written)) return false;
+            if (!NewLine.TryCopyTo(Free.Slice(written))) return false;
+
+            written += NewLine.Length;
+            return true;
+        }
+
+        private bool TryWriteLine(Utf8Span text, out int written)
+        {
+            if (!text.Bytes.TryCopyTo(Free))
+            {
+                written = 0;
+                return false;
+            }
+            written = text.Bytes.Length;
+            if (!NewLine.TryCopyTo(Free.Slice(written))) return false;
```

Comment:
Do we need to check that written is less than Free.Length? Otherwise the Slice could throw.

Reply:
Also, do we want written to be set to text.Bytes.Length if this TryCopyTo fails or should it be 0?

Reply:
yeah, good point.